src/main/java/org/dataone/hashstore/HashStore.java

```diff
@@ -73,66 +80,123 @@ public HashStore(Path storeDirectory)
      * @throws NullPointerException       Arguments are null for pid or object
      * @throws InterruptedException       Synchronization issue with objectLockedIds
      */
+    @Override
     public HashAddress storeObject(InputStream object, String pid, String additionalAlgorithm, String checksum,
             String checksumAlgorithm)
             throws NoSuchAlgorithmException, IOException, SecurityException, FileNotFoundException,
             FileAlreadyExistsException, IllegalArgumentException, NullPointerException, InterruptedException {
+        logHashStore.info("HashStore.storeObject - Called to store object for pid: " + pid);
+        // Begin input validation
         if (object == null) {
+            logHashStore.error("HashStore.storeObject - InputStream cannot be null, pid: " + pid);
             throw new NullPointerException("Invalid input stream, data is null.");
         }
-        if (pid == null || pid.isEmpty()) {
+        if (pid == null || pid.trim().isEmpty()) {
+            logHashStore.error("HashStore.storeObject - pid cannot be null or empty, pid: " + pid);
             throw new IllegalArgumentException("Pid cannot be null or empty, pid: " + pid);
         }
+        // Checksum cannot be empty or null if checksumAlgorithm is passed
+        if (checksumAlgorithm != null & checksum != null) {
+            if (checksum.trim().isEmpty()) {
+                logHashStore
+                        .error("HashStore.storeObject - checksum cannot be empty if checksumAlgorithm is supplied, pid: "
+                                + pid);
+                throw new IllegalArgumentException(
+                        "Checksum cannot be empty when checksumAlgorithm is supplied.");
```

Comment:
I have since refactored this input validation for checksum and checksumAlgorithm, and would need to update the method for a pid parameter that's only used in logs to incorporate this feedback - so for now I am going to leave it as is